rust_src/src/fonts.rs

```diff
@@ -129,4 +133,43 @@ pub fn font_match_p(spec: LispObject, font: LispObject) -> bool {
     unsafe { c_font_match_p(spec, font) }
 }
 
+/// Return a font-object for displaying a character at POSITION.
+/// Optional second arg WINDOW, if non-nil, is a window displaying
+/// the current buffer.  It defaults to the currently selected window.
+/// Optional third arg STRING, if non-nil, is a string containing the target
+/// character at index specified by POSITION.
+#[lisp_fn(min = "1", c_name = "font_at", name = "font-at")]
+pub fn font_at_lisp(
```

Comment:
`position` can be `LispNumber` no?

Reply:
`string` can be `Option<LispStringRef>` right?

Reply:
position can't be `LispNumber` because of the coerce marker check. But I have to move the `position.as_fixnum_or_error()`. But you're right with string.

Reply:
See #983

Reply:
Because of the C `font_at` we can leave this be for now. Eventually, we can use the right types.